### PR TITLE
DDF-4270 Remove Workspace Templates

### PR DESCRIPTION
**Pull request by willwill96**
#### What does this PR do?
	- Removes workspace templates from DDF workspace home page
	- Adds `Add New Workspace` button to allow creation of a blank workspace
#### Who is reviewing it? 
@djblue 
@rymach 
@GabrielFabian 
@gjvera 

#### Ask 2 committers to review/merge the PR and tag them here.
@bdeining
@djblue
#### How should this be tested?
1. Build and install
2. Navigate to `Workspaces` home page
3. Ensure that there is no dropdown in the top right for workspace templates
4. Create a Blank Workspace by clicking the `New Workspace` button
	- Ensure that the workspace does not run a search immediately
5. Mess with theme settings to ensure correct scaling of the `Create Workspace` area
6. Spot test workspace functionality and view logs for any potential regressions

#### What are the relevant tickets?
[DDF-4270](https://codice.atlassian.net/browse/DDF-4270)
#### Screenshots
Before:
image

After:
image

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/workspaces-templates/workspaces-templates.tsx [hunk 1]
```diff
@@ -15,26 +15,15 @@ import { CustomElement } from '../../styles/mixins'
...
   placeholder: string
-  createWorkspace: () => void
-  createLocalWorkspace: () => void
-  createAllWorkspace: () => void
-  createGeoWorkspace: () => void
-  createLatestWorkspace: () => void
 }
 
-const Root = styled<RootProps, 'div'>('div')`
+const Root = styled<null, 'div'>('div')`
```
Comment (willwill96) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/workspaces-templates/workspaces-templates.tsx]: I was not really sure the correct way to deal with this change

**Comment by gjvera** [comment]
might want to rebase with master, some of the commits you have in your pr are already in master

**Comment by willwill96** [comment]
@gjvera rebasing gave me merge conflicts, so i decided to do a merge instead (which didn't give conflicts)

**Comment by rymach** [comment]
is there a way for me to create a blank workspace? I used to use that option a lot

**Comment by willwill96** [comment]
@rymach There is not currently. Do you have any ideas for what that would look like with the removal of templates?

**Comment by gjvera** [comment]
@rymach @willwill96 If you just click on the search icon it creates an empty one for you.

**Comment by mojogitoverhere** [comment]
@gjvera 	That's not very user friendly

**Comment by willwill96** [comment]
@gjvera I think that actually creates a workspace with a `*` query and searches immediately

**Comment by TeresaHudson** [comment]
My suggestion would be to make workspaces consistent with our other card views and put an "add new workspace" button at the beginning of the workspaces area under the search. It would look the same as the button for creating new form templates, result sets, and AOIs.

**Comment by mojogitoverhere** [comment]
As an added bonus, @TeresaHudson 	's suggestion would also get us one step closer to the future Intrigue workspace layout

screen shot 2018-10-26 at 11 52 47 am

**Comment by willwill96** [comment]
Just updated the `After`picture - still working on getting the CSS right, but that is generally what I see it looking like at this point

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/workspace-create-item/workspace-create-item.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
+/**
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/workspace-create-item/workspace-create-item.tsx]: I'm not convinced this is necessary as a new component.  We could accomplish something very similar by using the existing Button component, in which case you might be able to use it directly where this component is used (or you could use this component to wrap the button component in the styles you need).

At the very least, it needs to be broken into a presentation component and a container component.

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/workspaces-items/workspaces-items.tsx [hunk 3]
```diff
@@ -63,6 +64,7 @@ const Root = styled.div`
 
 const WorkspaceItemRoot = styled.div`
   width: 18rem;
+  height: 8.5rem;
```
Comment (andrewkfiedler) [inline ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/workspaces-items/workspaces-items.tsx]: Why is this needed?

Comment (willwill96) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/workspaces-items/workspaces-items.tsx]: This was done to force the `div`'s for the `WorkspaceItem`s and the new creation button to have the same height

File: ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/card/card.tsx [hunk 4]
```diff
@@ -14,7 +14,7 @@ import styled from '../../styles/styled-components'
 
 const Root = styled.div`
   width: 100%;
-  height: auto;
```
[the remark this answers is not in the document]
Comment (willwill96) [reply ui/packages/catalog-ui-search/src/main/webapp/react-component/presentation/card/card.tsx]: This change ended up getting removed

**Comment by gjvera** [comment]
build now

**Comment by GabrielFabian** [comment]
hero build in progress

**Comment by GabrielFabian** [comment]
🎉 Hero build successful! 🎉 

- verified new button
- verified functionality of creating a new workspace
- verified when creating new workspace, a search is not run by default
- verified no regressions
- logged in as admin and as guest

LGTM!

screen shot 2018-10-30 at 5 18 20 pm
screen shot 2018-10-30 at 5 18 56 pm
